<a href="https://colab.research.google.com/github/kimkukhwa/Google-Data-Analytics-Capstone-Project/blob/main/google_data_analytics_capstone_bellabeat.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# [Bellabeat](https://bellabeat.com/): How Can a Wellness Technology Company Play It Smart?

## INTRODUCTION:
**About the company**

Urška Sršen and Sando Mur founded Bellabeat, a high-tech company that manufactures health-focused smart products.
Sršen used her background as an artist to develop beautifully designed technology that informs and inspires women around
the world. Collecting data on activity, sleep, stress, and reproductive health has allowed Bellabeat to empower women with
knowledge about their own health and habits. Since it was founded in 2013, Bellabeat has grown rapidly and quickly
positioned itself as a tech-driven wellness company for women.

By 2016, Bellabeat had opened offices around the world and launched multiple products. Bellabeat products became available
through a growing number of online retailers in addition to their own e-commerce channel on their website. The company
has invested in traditional advertising media, such as radio, out-of-home billboards, print, and television, but focuses on digital
marketing extensively. Bellabeat invests year-round in Google Search, maintaining active Facebook and Instagram pages, and
consistently engages consumers on Twitter. Additionally, Bellabeat runs video ads on Youtube and display ads on the Google
Display Network to support campaigns around key marketing dates.

**Questions**

Sršen asks you to analyze smart device usage data in order to gain insight into how consumers use non-Bellabeat smart
devices. She then wants you to select one Bellabeat product to apply these insights to in your presentation. These questions
will guide your analysis:
1. What are some trends in smart device usage?
2. How could these trends apply to Bellabeat customers?
3. How could these trends help influence Bellabeat marketing strategy?

You will produce a report with the following deliverables:
1. A clear summary of the business task
2. A description of all data sources used
3. Documentation of any cleaning or manipulation of data
4. A summary of your analysis
5. Supporting visualizations and key findings
6. Your top high-level content recommendations based on your analysis

In [69]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## STEP 1: ASK

1.1 Business Task:

    Analyze FitBit Fitness Tracker Data to gain insights into user behaviour and trend that influence Bellabeat marketing strategy
    
1.2 Key Stakeholders:

    1. Urška Sršen: Bellabeat’s cofounder and Chief Creative Officer
    2. Sando Mur: Mathematician, Bellabeat’s cofounder and key member of the Bellabeat executive team
    3. Bellabeat marketing analytics team: A team of data analysts guiding Bellabeat's marketing strategy.
    


## STEP 2: PREPARE

2.1 Description on Data Source:

    1. The data is publicly available on [Kaggle: FitBit Fitness Tracker Data](https://www.kaggle.com/datasets/arashnic/fitbit)
    2. This dataset generated by respondents to a distributed survey via Amazon Mechanical Turk between 03.12.2016-05.12.2016
    3. Thirty eligible Fitbit users consented to the submission of personal tracker data
    4. Individual reports can be parsed by export session ID (column A) or timestamp (column B)

2.2 Data Credibility & Limitations:

    1. Reliable - This data could have sample selection bias becasue it doesn't refelct the overall population
    2. Original - This data is third party information
    3. Comprehensive - This data includes all important information needed to answer the business question
    4. Current - The data is 6 years old
    5. Cited - Unknown
    
    Overall, the data source is evaluated as bad data, but it is not relevant at the moment since this is for the capstone project. 

2.3 Data Selection:
    
    dailyActivity_merged.csv



## STEP 3: PROCESS


3.1 Import Packages

In [70]:
import numpy as np 
import pandas as pd 
import matplotlib as plt
import plotly.express as px
import datetime as dt 

3.2 Import Data

In [71]:
df=pd.read_csv('/dailyActivity_merged.csv')



3.3 EDA

In [72]:
# Check the first 5 rows of dataframe
df.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

In [73]:
# Check the last 5 rows of dataframe
df.tail()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
935  8877689391     5/8/2016       10686       8.110000         8.110000   
936  8877689391     5/9/2016       20226      18.250000        18.250000   
937  8877689391    5/10/2016       10733       8.150000         8.150000   
938  8877689391    5/11/2016       21420      19.559999        19.559999   
939  8877689391    5/12/2016        8064       6.120000         6.120000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
935                       0.0                1.08                      0.20   
936                       0.0               11.10                      0.80   
937                       0.0                1.35                      0.46   
938                       0.0               13.22                      0.41   
939                       0.0                1.82                      0.04   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
935                 6.80                     0.00                 17   
936                 6.24                     0.05                 73   
937                 6.28                     0.00                 18   
938                 5.89                     0.00                 88   
939                 4.25                     0.00                 23   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
935                    4                   245              1174      2847  
936                   19                   217              1131      3710  
937                   11                   224              1187      2832  
938                   12                   213              1127      3832  
939                    1                   137               770      1849

In [74]:
#  Overview of the dataset, including the index dtype and column dtypes, non-null values, and memory usage
print(df.info())
print('rows x columns:',df.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

In [75]:
# Statistical summary for numerical columns
df.drop('Id',axis=1).describe()

TotalSteps  TotalDistance  TrackerDistance  LoggedActivitiesDistance  \
count    940.000000     940.000000       940.000000                940.000000   
mean    7637.910638       5.489702         5.475351                  0.108171   
std     5087.150742       3.924606         3.907276                  0.619897   
min        0.000000       0.000000         0.000000                  0.000000   
25%     3789.750000       2.620000         2.620000                  0.000000   
50%     7405.500000       5.245000         5.245000                  0.000000   
75%    10727.000000       7.712500         7.710000                  0.000000   
max    36019.000000      28.030001        28.030001                  4.942142   

       VeryActiveDistance  ModeratelyActiveDistance  LightActiveDistance  \
count          940.000000                940.000000           940.000000   
mean             1.502681                  0.567543             3.340819   
std              2.658941                  0.883580             2.040655   
min              0.000000                  0.000000             0.000000   
25%              0.000000                  0.000000             1.945000   
50%              0.210000                  0.240000             3.365000   
75%              2.052500                  0.800000             4.782500   
max             21.920000                  6.480000            10.710000   

       SedentaryActiveDistance  VeryActiveMinutes  FairlyActiveMinutes  \
count               940.000000         940.000000           940.000000   
mean                  0.001606          21.164894            13.564894   
std                   0.007346          32.844803            19.987404   
min                   0.000000           0.000000             0.000000   
25%                   0.000000           0.000000             0.000000   
50%                   0.000000           4.000000             6.000000   
75%                   0.000000          32.000000            19.000000   
max                   0.110000         210.000000           143.000000   

       LightlyActiveMinutes  SedentaryMinutes     Calories  
count            940.000000        940.000000   940.000000  
mean             192.812766        991.210638  2303.609574  
std              109.174700        301.267437   718.166862  
min                0.000000          0.000000     0.000000  
25%              127.000000        729.750000  1828.500000  
50%              199.000000       1057.500000  2134.000000  
75%              264.000000       1229.500000  2793.250000  
max              518.000000       1440.000000  4900.000000

In [76]:
# Check any na values
df.isna().sum()

Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
LoggedActivitiesDistance    0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

In [77]:
# Unique entries over columns and rows in the object
df.nunique()

Id                           33
ActivityDate                 31
TotalSteps                  842
TotalDistance               615
TrackerDistance             613
LoggedActivitiesDistance     19
VeryActiveDistance          333
ModeratelyActiveDistance    211
LightActiveDistance         491
SedentaryActiveDistance       9
VeryActiveMinutes           122
FairlyActiveMinutes          81
LightlyActiveMinutes        335
SedentaryMinutes            549
Calories                    734
dtype: int64

In [78]:
# Check if every user tracked their activity everyday
df.groupby('ActivityDate')[['Id']].nunique()

Id
ActivityDate    
4/12/2016     33
4/13/2016     33
4/14/2016     33
4/15/2016     33
4/16/2016     32
4/17/2016     32
4/18/2016     32
4/19/2016     32
4/20/2016     32
4/21/2016     32
4/22/2016     32
4/23/2016     32
4/24/2016     32
4/25/2016     32
4/26/2016     32
4/27/2016     32
4/28/2016     32
4/29/2016     32
4/30/2016     31
5/1/2016      30
5/10/2016     26
5/11/2016     24
5/12/2016     21
5/2/2016      29
5/3/2016      29
5/4/2016      29
5/5/2016      29
5/6/2016      29
5/7/2016      29
5/8/2016      27
5/9/2016      27

In [79]:
# Check who (Id) missed daily tracking
df.groupby('Id')[['ActivityDate']].nunique()

ActivityDate
Id                      
1503960366            31
1624580081            31
1644430081            30
1844505072            31
1927972279            31
2022484408            31
2026352035            31
2320127002            31
2347167796            18
2873212765            31
3372868164            20
3977333714            30
4020332650            31
4057192912             4
4319703577            31
4388161847            31
4445114986            31
4558609924            31
4702921684            31
5553957443            31
5577150313            30
6117666160            28
6290855005            29
6775888955            26
6962181067            31
7007744171            26
7086361926            31
8053475328            31
8253242879            19
8378563200            31
8583815059            31
8792009665            29
8877689391            31

In [80]:
# Check observation for Id=4057192912
df[df.Id==4057192912]

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
377  4057192912    4/12/2016        5394           4.03             4.03   
378  4057192912    4/13/2016        5974           4.47             4.47   
379  4057192912    4/14/2016           0           0.00             0.00   
380  4057192912    4/15/2016        3984           2.95             2.95   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
377                       0.0                0.00                      0.00   
378                       0.0                0.00                      0.00   
379                       0.0                0.00                      0.00   
380                       0.0                0.21                      0.26   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
377                 3.94                      0.0                  0   
378                 4.37                      0.0                  0   
379                 0.00                      0.0                  0   
380                 2.44                      0.0                  3   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
377                    0                   164              1276      2286  
378                    0                   160              1280      2306  
379                    0                     0              1440      1776  
380                    6                    88               873      1527

In [81]:
# Check where records are 0
df.loc[(df==0).any(axis=1)]

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0    1503960366    4/12/2016       13162       8.500000         8.500000   
1    1503960366    4/13/2016       10735       6.970000         6.970000   
2    1503960366    4/14/2016       10460       6.740000         6.740000   
3    1503960366    4/15/2016        9762       6.280000         6.280000   
4    1503960366    4/16/2016       12669       8.160000         8.160000   
..          ...          ...         ...            ...              ...   
935  8877689391     5/8/2016       10686       8.110000         8.110000   
936  8877689391     5/9/2016       20226      18.250000        18.250000   
937  8877689391    5/10/2016       10733       8.150000         8.150000   
938  8877689391    5/11/2016       21420      19.559999        19.559999   
939  8877689391    5/12/2016        8064       6.120000         6.120000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                         0.0                1.88                      0.55   
1                         0.0                1.57                      0.69   
2                         0.0                2.44                      0.40   
3                         0.0                2.14                      1.26   
4                         0.0                2.71                      0.41   
..                        ...                 ...                       ...   
935                       0.0                1.08                      0.20   
936                       0.0               11.10                      0.80   
937                       0.0                1.35                      0.46   
938                       0.0               13.22                      0.41   
939                       0.0                1.82                      0.04   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                   6.06                     0.00                 25   
1                   4.71                     0.00                 21   
2                   3.91                     0.00                 30   
3                   2.83                     0.00                 29   
4                   5.04                     0.00                 36   
..                   ...                      ...                ...   
935                 6.80                     0.00                 17   
936                 6.24                     0.05                 73   
937                 6.28                     0.00                 18   
938                 5.89                     0.00                 88   
939                 4.25                     0.00                 23   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                     13                   328               728      1985  
1                     19                   217               776      1797  
2                     11                   181              1218      1776  
3                     34                   209               726      1745  
4                     10                   221               773      1863  
..                   ...                   ...               ...       ...  
935                    4                   245              1174      2847  
936                   19                   217              1131      3710  
937                   11                   224              1187      2832  
938                   12                   213              1127      3832  
939                    1                   137               770      1849  

[937 rows x 15 columns]

In [82]:
# Check correlation of all columns (Any missing values, non-numeric are automatically excluded)
df.drop('Id',axis=1).corr( )

TotalSteps  TotalDistance  TrackerDistance  \
TotalSteps                  1.000000       0.985369         0.984822   
TotalDistance               0.985369       1.000000         0.999505   
TrackerDistance             0.984822       0.999505         1.000000   
LoggedActivitiesDistance    0.181849       0.188332         0.162585   
VeryActiveDistance          0.740115       0.794582         0.794338   
ModeratelyActiveDistance    0.507105       0.470758         0.470277   
LightActiveDistance         0.692208       0.662002         0.661365   
SedentaryActiveDistance     0.070505       0.082389         0.074591   
VeryActiveMinutes           0.667079       0.681297         0.680816   
FairlyActiveMinutes         0.498693       0.462899         0.463154   
LightlyActiveMinutes        0.569600       0.516300         0.514713   
SedentaryMinutes           -0.327484      -0.288094        -0.289343   
Calories                    0.591568       0.644962         0.645313   

                          LoggedActivitiesDistance  VeryActiveDistance  \
TotalSteps                                0.181849            0.740115   
TotalDistance                             0.188332            0.794582   
TrackerDistance                           0.162585            0.794338   
LoggedActivitiesDistance                  1.000000            0.150852   
VeryActiveDistance                        0.150852            1.000000   
ModeratelyActiveDistance                  0.076527            0.192986   
LightActiveDistance                       0.138302            0.157669   
SedentaryActiveDistance                   0.154996            0.046117   
VeryActiveMinutes                         0.234443            0.826681   
FairlyActiveMinutes                       0.053860            0.211730   
LightlyActiveMinutes                      0.102135            0.059845   
SedentaryMinutes                         -0.046999           -0.061754   
Calories                                  0.207595            0.491959   

                          ModeratelyActiveDistance  LightActiveDistance  \
TotalSteps                                0.507105             0.692208   
TotalDistance                             0.470758             0.662002   
TrackerDistance                           0.470277             0.661365   
LoggedActivitiesDistance                  0.076527             0.138302   
VeryActiveDistance                        0.192986             0.157669   
ModeratelyActiveDistance                  1.000000             0.237847   
LightActiveDistance                       0.237847             1.000000   
SedentaryActiveDistance                   0.005793             0.099503   
VeryActiveMinutes                         0.225464             0.154966   
FairlyActiveMinutes                       0.946934             0.220129   
LightlyActiveMinutes                      0.162092             0.885697   
SedentaryMinutes                         -0.221436            -0.413552   
Calories                                  0.216790             0.466917   

                          SedentaryActiveDistance  VeryActiveMinutes  \
TotalSteps                               0.070505           0.667079   
TotalDistance                            0.082389           0.681297   
TrackerDistance                          0.074591           0.680816   
LoggedActivitiesDistance                 0.154996           0.234443   
VeryActiveDistance                       0.046117           0.826681   
ModeratelyActiveDistance                 0.005793           0.225464   
LightActiveDistance                      0.099503           0.154966   
SedentaryActiveDistance                  1.000000           0.008258   
VeryActiveMinutes                        0.008258           1.000000   
FairlyActiveMinutes                     -0.022361           0.312420   
LightlyActiveMinutes                     0.124185           0.051926   
SedentaryMinutes                         0.035475          -0.164

In [83]:
# Check records by Unique Key (Id+Timestamp)
df.groupby(['Id','ActivityDate']).sum()


TotalSteps  TotalDistance  TrackerDistance  \
Id         ActivityDate                                               
1503960366 4/12/2016          13162           8.50             8.50   
           4/13/2016          10735           6.97             6.97   
           4/14/2016          10460           6.74             6.74   
           4/15/2016           9762           6.28             6.28   
           4/16/2016          12669           8.16             8.16   
...                             ...            ...              ...   
8877689391 5/5/2016           14055          10.67            10.67   
           5/6/2016           21727          19.34            19.34   
           5/7/2016           12332           8.13             8.13   
           5/8/2016           10686           8.11             8.11   
           5/9/2016           20226          18.25            18.25   

                         LoggedActivitiesDistance  VeryActiveDistance  \
Id         ActivityDate                                                 
1503960366 4/12/2016                          0.0                1.88   
           4/13/2016                          0.0                1.57   
           4/14/2016                          0.0                2.44   
           4/15/2016                          0.0                2.14   
           4/16/2016                          0.0                2.71   
...                                           ...                 ...   
8877689391 5/5/2016                           0.0                5.46   
           5/6/2016                           0.0               12.79   
           5/7/2016                           0.0                0.08   
           5/8/2016                           0.0                1.08   
           5/9/2016                           0.0               11.10   

                         ModeratelyActiveDistance  LightActiveDistance  \
Id         ActivityDate                                                  
1503960366 4/12/2016                         0.55                 6.06   
           4/13/2016                         0.69                 4.71   
           4/14/2016                         0.40                 3.91   
           4/15/2016                         1.26                 2.83   
           4/16/2016                         0.41                 5.04   
...                                           ...                  ...   
8877689391 5/5/2016                          0.82                 4.37   
           5/6/2016                          0.29                 6.16   
           5/7/2016                          0.96                 6.99   
           5/8/2016                          0.20                 6.80   
           5/9/2016                          0.80                 6.24   

                         SedentaryActiveDistance  VeryActiveMinutes  \
Id         ActivityDate                                               
1503960366 4/12/2016                        0.00                 25   
           4/13/2016                        0.00                 21   
           4/14/2016                        0.00                 30   
           4/15/2016                        0.00                 29   
           4/16/2016                        0.00                 36   
...                                          ...                ...   
8877689391 5/5/2016                         0.00                 67   
           5/6/2016                         0.00                 96   
           5/7/2016                         0.00                105   
           5/8/2016                         0.00                 17   
           5/9/2016                         0.05                 73   

                         FairlyActiveMinutes  LightlyActiveMinutes  \
Id         ActivityDate                                              
1503960366 4/12/2016                      13                   328   
           4/13/2016                      19          

In [84]:
# Data sanity check: TotalDistance
total=df.filter(regex='Distance')
total['sum']=total.iloc[:,2:].sum(axis=1).round(1)
total

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:3: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



TotalDistance  TrackerDistance  LoggedActivitiesDistance  \
0         8.500000         8.500000                       0.0   
1         6.970000         6.970000                       0.0   
2         6.740000         6.740000                       0.0   
3         6.280000         6.280000                       0.0   
4         8.160000         8.160000                       0.0   
..             ...              ...                       ...   
935       8.110000         8.110000                       0.0   
936      18.250000        18.250000                       0.0   
937       8.150000         8.150000                       0.0   
938      19.559999        19.559999                       0.0   
939       6.120000         6.120000                       0.0   

     VeryActiveDistance  ModeratelyActiveDistance  LightActiveDistance  \
0                  1.88                      0.55                 6.06   
1                  1.57                      0.69                 4.71   
2                  2.44                      0.40                 3.91   
3                  2.14                      1.26                 2.83   
4                  2.71                      0.41                 5.04   
..                  ...                       ...                  ...   
935                1.08                      0.20                 6.80   
936               11.10                      0.80                 6.24   
937                1.35                      0.46                 6.28   
938               13.22                      0.41                 5.89   
939                1.82                      0.04                 4.25   

     SedentaryActiveDistance   sum  
0                       0.00   8.5  
1                       0.00   7.0  
2                       0.00   6.8  
3                       0.00   6.2  
4                       0.00   8.2  
..                       ...   ...  
935                     0.00   8.1  
936                     0.05  18.2  
937                     0.00   8.1  
938                     0.00  19.5  
939                     0.00   6.1  

[940 rows x 8 columns]

### EDA Summary:
<font color='red'>
    
    1. 33 Unique Ids instead of 30 Unique Ids
    2. ActivityDate dtype needs to be converted to datetime dtype for accurate sorting/filtering purposes 
    3. No Null or Missing values, but it is observed that users did not have to track their activity daily and some records are "0" 
    4. Add Total Active Minutes

3.4 Data Transformation

In [85]:
# ActivityDate to datetime
df['ActivityDate']=pd.to_datetime(df['ActivityDate'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   Id                        940 non-null    int64         
 1   ActivityDate              940 non-null    datetime64[ns]
 2   TotalSteps                940 non-null    int64         
 3   TotalDistance             940 non-null    float64       
 4   TrackerDistance           940 non-null    float64       
 5   LoggedActivitiesDistance  940 non-null    float64       
 6   VeryActiveDistance        940 non-null    float64       
 7   ModeratelyActiveDistance  940 non-null    float64       
 8   LightActiveDistance       940 non-null    float64       
 9   SedentaryActiveDistance   940 non-null    float64       
 10  VeryActiveMinutes         940 non-null    int64         
 11  FairlyActiveMinutes       940 non-null    int64         
 12  LightlyActiveMinutes  

In [86]:
# Add TotalActiveMinutes
min_col=df.filter(regex='Minutes')
df['TotalActiveMinutes']=min_col.iloc[:,:].sum(axis=1)
df.head()


Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366   2016-04-12       13162           8.50             8.50   
1  1503960366   2016-04-13       10735           6.97             6.97   
2  1503960366   2016-04-14       10460           6.74             6.74   
3  1503960366   2016-04-15        9762           6.28             6.28   
4  1503960366   2016-04-16       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   13                   328               728      1985   
1                   19                   217               776      1797   
2                   11                   181              1218      1776   
3                   34                   209               726      1745   
4                   10                   221               773      1863   

   TotalActiveMinutes  
0                1094  
1                1033  
2                1440  
3                 998  
4                1040

## STEP 4: ANALYZE

4.1 User Activity Trend

In [87]:
df.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108171            1.502681                  0.567543   
std                    0.619897            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.942142           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

          Calories  TotalActiveMinutes  
count   940.000000          940.000000  
mean   2303.609574         1218.753191  
std     718.166862          265.931767  
min       0.000000            2.000000  
25%    1828.500000          989.750000  
50%    2134.000000         1440.000000  
75%    2793.250000         1440.000000  
max    4900.000000         1440.000000

In [88]:
df['ActivityDay']=df['ActivityDate'].dt.day_name()
df.groupby(['ActivityDay'])['Id'].count()


ActivityDay
Friday       126
Monday       120
Saturday     124
Sunday       121
Thursday     147
Tuesday      152
Wednesday    150
Name: Id, dtype: int64

In [89]:
df.groupby(['ActivityDay'])['TotalSteps','TotalDistance','TotalActiveMinutes','Calories'].mean().sort_values(['TotalSteps','TotalDistance','TotalActiveMinutes','Calories'])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



TotalSteps  TotalDistance  TotalActiveMinutes     Calories
ActivityDay                                                             
Sunday       6933.231405       5.027190         1198.743802  2263.000000
Thursday     7405.836735       5.312245         1178.782313  2199.571429
Friday       7448.230159       5.309921         1236.674603  2331.785714
Wednesday    7559.373333       5.488333         1213.213333  2302.620000
Monday       7780.866667       5.552917         1257.108333  2324.208333
Tuesday      8125.006579       5.832237         1241.993421  2356.013158
Saturday     8152.975806       5.854677         1208.548387  2354.967742

**Basic Statistical Summary**:
- On average, users performed 7638 steps, moved 5.5km distance, and burned 2304 calories per day. [Our average users are Somewhat active (7,500 to 9,999 steps per day)](https://www.10000steps.org.au/articles/healthy-lifestyles/counting-steps/#:~:text=Low%20active%20is%205%2C000%20to,active%20is%20more%20than%2012%2C500).
- Users are active for 20 hours (1218.7) per day, on average,including from sedentary level to very active level.  
- User used the device most on Tuesday, but performed most steps and moved most distance on Saturday. 

4.2 User Logging Trend

In [90]:
df['LoggedDistince%']=df['LoggedActivitiesDistance']/df['TrackerDistance']
df[df['LoggedActivitiesDistance']>0][['TrackerDistance', 'LoggedActivitiesDistance','LoggedDistince%']].head()

TrackerDistance  LoggedActivitiesDistance  LoggedDistince%
668             5.27                  1.959596         0.371840
689             7.88                  4.081692         0.517981
693             9.08                  2.785175         0.306737
707             8.68                  3.167822         0.364956
711             9.48                  4.869783         0.513690

In [91]:
no_log=df[df['LoggedActivitiesDistance']==0]
no_log.shape

(908, 18)

In [92]:
logged=df[df['LoggedActivitiesDistance']>0]
logged.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  3.200000e+01     32.000000      32.000000        32.000000   
mean   7.681637e+09  12042.500000       9.147188         8.725625   
std    7.092501e+08   3382.868224       2.377527         2.132471   
min    6.775889e+09   6064.000000       4.810000         4.810000   
25%    7.007744e+09   9035.250000       7.165000         7.165000   
50%    7.693154e+09  12633.500000       9.690000         9.070000   
75%    8.378563e+09  14177.500000      10.622500         9.942500   
max    8.378563e+09  20067.000000      14.300000        13.420000   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                 32.000000            32.00000                 32.000000   
mean                   3.177521             3.57375                  0.869687   
std                    1.253865             1.68858                  0.522485   
min                    1.959596             0.63000                  0.160000   
25%                    2.092147             2.06000                  0.582500   
50%                    2.253081             4.29000                  0.705000   
75%                    4.863790             4.67750                  1.037500   
max                    4.942142             6.90000                  2.120000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count            32.000000                32.000000          32.000000   
mean              4.690937                 0.007188          71.187500   
std               1.246710                 0.025932          28.723866   
min               0.730000                 0.000000          34.000000   
25%               3.987500                 0.000000          53.000000   
50%               4.500000                 0.000000          63.000000   
75%               5.295000                 0.000000          76.000000   
max               7.950000                 0.110000         137.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count            32.000000             32.000000         32.000000   
mean             17.843750            227.781250        870.593750   
std               9.615543             85.253601        179.824441   
min               3.000000             47.000000        607.000000   
25%              11.500000            164.250000        722.250000   
50%              16.000000            204.500000        812.000000   
75%              23.250000            304.000000       1028.250000   
max              42.000000            382.000000       1321.000000   

          Calories  TotalActiveMinutes  LoggedDistince%  
count    32.000000           32.000000        32.000000  
mean   3305.000000         1187.406250         0.368726  
std     643.527827          215.342766         0.115856  
min    2105.000000          902.000000         0.187600  
25%    2825.500000         1015.500000         0.286730  
50%    3335.500000         1060.500000         0.368899  
75%    3787.250000         1440.000000         0.476286  
max    4236.000000         1440.000000         0.564105

In [93]:
logged['Id'].nunique()

4

In [94]:
df['LoggedYN'] = df['LoggedActivitiesDistance'].apply(lambda x: "Y" if x>0 else "N")
df[df['LoggedActivitiesDistance']>0].head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
668  6775888955   2016-04-26        7091           5.27             5.27   
689  6962181067   2016-04-21       11835           9.71             7.88   
693  6962181067   2016-04-25       13239           9.27             9.08   
707  6962181067   2016-05-09       12342           8.72             8.68   
711  7007744171   2016-04-12       14172          10.29             9.48   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
668                  1.959596                3.48                      0.87   
689                  4.081692                3.99                      2.10   
693                  2.785175                3.02                      1.68   
707                  3.167822                3.90                      1.18   
711                  4.869783                4.50                      0.38   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
668                 0.73                     0.00                 42   
689                 3.51                     0.11                 53   
693                 4.46                     0.10                 35   
707                 3.65                     0.00                 43   
711                 5.41                     0.00                 53   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
668                   30                    47              1321      2584   
689                   27                   214               708      2179   
693                   31                   282               637      2194   
707                   21                   231               607      2105   
711                    8                   355              1024      2937   

     TotalActiveMinutes ActivityDay  LoggedDistince% LoggedYN  
668                1440     Tuesday         0.371840        Y  
689                1002    Thursday         0.517981        Y  
693                 985      Monday         0.306737        Y  
707                 902      Monday         0.364956        Y  
711                1440     Tuesday         0.513690        Y

**Logging Activity Trend:**
- 908 entires are not logged, only 32 entries(4 users) are logged their Activities Distance. 
- When logged, Activities Distance is about 37% of their TrackerDistance on average.
- Users, who logged Activities Distance, burned 3305 calories, peformed 12,043 steps, and moved 9.15km per day on average. [Those users are Active~Highly Active ](https://www.10000steps.org.au/articles/healthy-lifestyles/counting-steps/#:~:text=Low%20active%20is%205%2C000%20to,active%20is%20more%20than%2012%2C500)


## STEP 5: SHARE

5.1 Visualization


In [95]:
#import matplotlib & plotly
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.offline as py
from plotly.offline import init_notebook_mode, iplot
init_notebook_mode(connected=True)

a. User Activity Tracking Trend -Histogram

In [96]:
fig = px.histogram(df, x='ActivityDay')
fig.show(renderer="colab")


In [97]:
fig = px.histogram(df, x='ActivityDate')
fig.update_layout(bargap=0.2)
fig.show(renderer="colab")

b. User Activity Tracking - Box plot

In [98]:
avg_steps_byday=df.groupby(['ActivityDay'])['TotalSteps','TotalDistance','Calories','TotalActiveMinutes'].mean()
avg_steps_byday=avg_steps_byday.reset_index()
avg_steps_byday

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



ActivityDay   TotalSteps  TotalDistance     Calories  TotalActiveMinutes
0      Friday  7448.230159       5.309921  2331.785714         1236.674603
1      Monday  7780.866667       5.552917  2324.208333         1257.108333
2    Saturday  8152.975806       5.854677  2354.967742         1208.548387
3      Sunday  6933.231405       5.027190  2263.000000         1198.743802
4    Thursday  7405.836735       5.312245  2199.571429         1178.782313
5     Tuesday  8125.006579       5.832237  2356.013158         1241.993421
6   Wednesday  7559.373333       5.488333  2302.620000         1213.213333

In [106]:

fig = px.box(df, x="ActivityDay", y="TotalSteps",color="LoggedYN", title="Steps taken by Day")
fig.show(renderer="colab")


In [107]:
fig = px.box(df, x="ActivityDay", y="TotalDistance",color="LoggedYN", title="Distance moved by Day")
fig.show(renderer="colab")

In [108]:
fig = px.box(df, x="ActivityDay", y="Calories",color="LoggedYN", title="Calories burned by Day")
fig.show(renderer="colab")

c. User Activity Tracking - Pie Chart

In [109]:
#import Go
import plotly.graph_objects as go
# Calculation for pie chart
very_active_mins = df["VeryActiveMinutes"].mean().round(1)
fairly_active_mins = df["FairlyActiveMinutes"].mean().round(1)
lightly_active_mins = df["LightlyActiveMinutes"].mean().round(1)
sedentary_mins = df["SedentaryMinutes"].mean().round(1)
labels = ['very_active', 'fairly_active', 'lightly_active', 'sedentary']
values = [very_active_mins, fairly_active_mins, lightly_active_mins, sedentary_mins]

# plot pie chart
fig = go.Figure(data=[go.Pie(labels=labels, values=values)])
colors = ['lightgreen', 'mediumturquoise', 'darkorange', 'gold']
fig.update_traces(hoverinfo='label+percent', textinfo='percent', textfont_size=20,
                  marker=dict(colors=colors, line=dict(color='#000000', width=2)))
fig.show(renderer="colab")

d. User Activity Tracking - Heatmap & 

In [103]:
heatmap_df=df[["TotalSteps","TotalDistance","TotalActiveMinutes","Calories","LoggedYN"]]
heatmap_df

TotalSteps  TotalDistance  TotalActiveMinutes  Calories LoggedYN
0         13162       8.500000                1094      1985        N
1         10735       6.970000                1033      1797        N
2         10460       6.740000                1440      1776        N
3          9762       6.280000                 998      1745        N
4         12669       8.160000                1040      1863        N
..          ...            ...                 ...       ...      ...
935       10686       8.110000                1440      2847        N
936       20226      18.250000                1440      3710        N
937       10733       8.150000                1440      2832        N
938       21420      19.559999                1440      3832        N
939        8064       6.120000                 931      1849        N

[940 rows x 5 columns]

In [110]:
fig = px.scatter_matrix(heatmap_df, color="LoggedYN")
fig.show(renderer="colab")

In [111]:
fig = px.imshow(df.iloc[:,1:-2].corr())
fig.show(renderer="colab")

**Insight from Visualization**
- Usage of the smart device is higher during mid-week from Tuesday to Friday.
- During the mid-week, users take more steps, move more distances, and burn more calories. Very active users do not track their activities during the weekend.
- 81% of Total Active Minutes is classified as Sedentary level. The heatmap also observed that there is no correlation between the duration of smart device usage and a more active lifestyle. Moreover, only four users (out of 33) logged their Activity Distance; it can be assumed that the smart device is used in daily life for most people rather than being used particularly to track work-out/calories burned. 
- Scatter plot and heatmap confirm that the more steps users take and the distance they move, the more calories they burn.


## STEP 6: ACT

ANSWER THE BUSINESS QUESTIONS:

1. What are the trends identified?

- Usage of the smart device is higher during mid-week from Tuesday to Friday.
- The smart device is used in daily life rather than being used to track specific fitness activities.
- The average users are ‘somewhat active’ based on their average daily activity. 


2. How could these trends apply to Bellabeat customers?

- Bellabeat customers are more open to fitness ideas that they can practice during the week, during lunch hour, after work, or before going to work.
- Various Mid to low-level activities can be helpful for the customers to gradually/gently increase their daily activity levels and active times.
- Notification to log their activities can encourage their habits to track their fitness activities/routine.  


3. How could these trends help influence Bellabeat marketing strategy?

To target the majority of users:

- In terms of product options and design, Bellabeat customers prefer a wearable they can use daily.
- In terms of an app push notification, the customers will be interested in learning various low-mid level daily activities they can try during the weekdays.
- In the app, the customers want to see more low-level activity fitness tracking options such as meditation, breathing exercises and stretching.

To target the highly active users:

- In terms of product options, highly active customers will prefer a product/services that can track their progress, take on challenges, and compete with other users.
- In terms of app push notifications, the customers will be interested in learning how to improve their performance, specific techniques, and professional community in a local area.
- In the app, the customers want to see their quality of break (or sleep).